src/test/java/teammates/test/driver/MockHttpServletRequest.java

```diff
@@ -0,0 +1,352 @@
+package teammates.test.driver;
+
+import java.io.BufferedReader;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+/**
+ * Mocks {@link HttpServletRequest} for testing purpose.
+ *
+ * <p>Only important methods are modified here; everything else are auto-generated.
+ */
+public class MockHttpServletRequest implements HttpServletRequest {
```

Comment:
Can `HttpUnit` do the mocking?

Reply:
Yes for Servlets, no for Filters, so it's a no.